mfr/extensions/tabular/libs/stdlib_tools.py

```diff
@@ -30,3 +34,21 @@ def csv_stdlib(fp):
         })
     rows = [row for row in reader]
     return columns, rows
+
+
+def detect_quoted_fields(dialect, data):
+    if dialect != csv.excel:
+        if dialect.quotechar == '"':
+            if re.search('\'[[({]\".+\",', data):
+                dialect.quotechar = "'"
+            if re.search('\'\'\'[[({]\".+\",', data):
+                dialect.doublequote = True
+            return dialect
```

Comment:
These return statements are superfluous, with a return at the end of the function.